Gems/ROS2/Code/Source/Camera/PostProcessing/ROS2ImageEncodingConversionComponent.cpp

```diff
@@ -0,0 +1,206 @@
+        }
+
+        const AZStd::unordered_map<EncodingConversion, const AZStd::function<void(sensor_msgs::msg::Image&)>> supportedFormatChange = {
+            { { ImageEncoding::RGBA8, ImageEncoding::RGB8 }, Rgba8ToRgb8 },
+        };
+
+        AZ::Outcome<void, AZStd::string> ValidateEncodingConversion(EncodingConversion newConversion)
+        {
+            if (newConversion.encodingIn == newConversion.encodingOut)
+            {
+                return AZ::Failure(AZStd::string("Conversion to same type if forbidden"));
+            }
+            if (supportedFormatChange.find(newConversion) == supportedFormatChange.end())
+            {
+                return AZ::Failure(AZStd::string::format(
+                    "Unsupported encoding change from %s to %s",
+                    ImageEncodingNames.at(newConversion.encodingIn),
+                    ImageEncodingNames.at(newConversion.encodingOut)));
+            }
+            return AZ::Success();
+        }
+
+    } // namespace
+
+    void EncodingConversion::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EncodingConversion>()
+                ->Version(0)
+                ->Field("EncodingIn", &EncodingConversion::encodingIn)
+                ->Field("EncodingOut", &EncodingConversion::encodingOut);
+
+            if (AZ::EditContext* ec = serializeContext->GetEditContext())
+            {
+                ec->Class<EncodingConversion>("Encoding Conversion", "Specifies encoding conversion")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::ComboBox, &EncodingConversion::encodingIn, "Encoding In", "Encoding of the input image")
+                    ->EnumAttribute(ImageEncoding::RGBA8, "rgba8")
+                    ->EnumAttribute(ImageEncoding::RGB8, "rgb8")
+                    ->EnumAttribute(ImageEncoding::Mono8, "mono8")
+                    ->EnumAttribute(ImageEncoding::Mono16, "mono16")
+                    ->Attribute(AZ::Edit::Attributes::ChangeValidate, &EncodingConversion::ValidateInputEncoding)
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::ComboBox, &EncodingConversion::encodingOut, "Encoding Out", "Encoding of the output image")
+                    ->EnumAttribute(ImageEncoding::RGBA8, "rgba8")
+                    ->EnumAttribute(ImageEncoding::RGB8, "rgb8")
+                    ->EnumAttribute(ImageEncoding::Mono8, "mono8")
+                    ->EnumAttribute(ImageEncoding::Mono16, "mono16")
+                    ->Attribute(AZ::Edit::Attributes::ChangeValidate, &EncodingConversion::ValidateOutputEncoding);
+            }
+        }
+    }
+
+    AZ::Outcome<void, AZStd::string> EncodingConversion::ValidateInputEncoding(void* newValue, const AZ::Uuid& valueType)
+    {
+        AZ_TracePrintf("EncodingConversion", "ValidateInputEncodingConversion");
```

Comment:
`AZ_TracePrintf` is deprecated, use AZ_Trace or AZ_Info instead (see https://github.com/o3de/o3de/issues/14838 )